### PR TITLE
[DO NOT CLOSE] Added functionality and tests to delete multiple environments

### LINKED ISSUE #1376: Remove multiple environments
## Description
At the moment, running `Delete Environment` from the Command Palette only allows you to select one environment to remove.
We should probably make this more consistent with Removing Packages, and allow a user to delete multiple at once.
## Steps to Reproduce

1.
2.
3.

## Existing issues

- [ ] [Stack Overflow issues](https://stackoverflow.com/questions/tagged/ibp-vscode-extension)
- [ ] [GitHub Issues](https://github.com/IBM-Blockchain/blockchain-vscode-extension/issues)

## Your Environment

* IBM Blockchain Platform Extension Version:
* Visual Studio Code version:
* Environment name and version (e.g. Node.js v8.12.0, npm v6.4.1):
* Operating System and version:
* Link to your project:

### PR DESCRIPTION
**Pull request by toggle007**
Added tests and functionality to allow multiple environments to be deleted.
To test this: you can add multiple environments and use the command palette to delete all of the newly added environments (excluding the local fabric) at once.
Furthermore, unlike before, when deleting an environment, it is also disconnected from the environment when deleted.

Closes #1376 
Signed-off-by: Akshat Shah <akshatshah@akshats-mbp.hursley.uk.ibm.com>

### CODE REVIEW

File: test/commands/deleteEnvironmentCommand.test.ts [hunk 1]
```diff
@@ -105,7 +105,42 @@ describe('DeleteEnvironmentCommand', () => {
...
+
+            environments = vscode.workspace.getConfiguration().get(SettingConfigurations.FABRIC_ENVIRONMENTS);
+            environments.length.should.equal(0);
+
+            logSpy.getCall(0).should.have.been.calledWithExactly(LogType.INFO, undefined, `delete environment`);
+            logSpy.getCall(1).should.have.been.calledWithExactly(LogType.SUCCESS, `Successfully deleted environments`);
+        });
+
+        it('should test that the only non-local environment can be deleted from the command', async () => {
+            await vscode.workspace.getConfiguration().update(SettingConfigurations.FABRIC_ENVIRONMENTS, [{name: 'myenvironmentA'}], vscode.ConfigurationTarget.Global);
```
Comment (cazfletch) [inline test/commands/deleteEnvironmentCommand.test.ts]: There is already a test to make sure you can't delete the local environment, all this test does is test that you can delete a non local one which we have already????

File: extension/commands/UserInputUtil.ts [hunk 2]
```diff
@@ -150,12 +150,6 @@ export class UserInputUtil {
             return { label: label, data: environment };
         });
 
-        if (environmentsQuickPickItems.length === 1) {
```
[the remark this answers is not in the document]
Comment (toggle007) [reply extension/commands/UserInputUtil.ts]: Fixed

File: extension/commands/UserInputUtil.ts [hunk 3]
```diff
@@ -150,12 +150,6 @@ export class UserInputUtil {
...
-            return environmentsQuickPickItems[0];
-        } else if (environmentsQuickPickItems.length === 0) {
```
[the remark this answers is not in the document]
Comment (toggle007) [reply extension/commands/UserInputUtil.ts]: Re-added - was a slight confusion

File: extension/commands/deleteEnvironmentCommand.ts [hunk 4]
```diff
@@ -42,40 +42,52 @@ export async function deleteEnvironment(environment: FabricEnvironmentTreeItem |
...
-
         const extDir: string = vscode.workspace.getConfiguration().get(SettingConfigurations.EXTENSION_DIRECTORY);
         const homeExtDir: string = UserInputUtil.getDirPath(extDir);
-        const environmentPath: string = path.join(homeExtDir, 'environments', environmentRegistryEntry.name);
-        await fs.remove(environmentPath);
 
-        await FabricEnvironmentRegistry.instance().delete(environmentRegistryEntry.name);
-        outputAdapter.log(LogType.SUCCESS, `Successfully deleted ${environmentRegistryEntry.name} environment`);
+        for (const _environment of environmentsToDelete) {
+            const connectedRegistry: FabricEnvironmentRegistryEntry = FabricEnvironmentManager.instance().getEnvironmentRegistryEntry();
```
[the remark this answers is not in the document]
Comment (toggle007) [reply extension/commands/deleteEnvironmentCommand.ts]: Fixed

**Pull request closed without merging**